src/AspNet.Security.OAuth.Shopify/ShopifyAuthenticationHandler.cs

```diff
@@ -0,0 +1,218 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Shopify
+{
+    /// <inheritdoc />
+    [UsedImplicitly]
+    public class ShopifyAuthenticationHandler : OAuthHandler<ShopifyAuthenticationOptions>
+    {
+        /// <inheritdoc />
+        public ShopifyAuthenticationHandler(
+            [NotNull] IOptionsMonitor<ShopifyAuthenticationOptions> options, 
+            [NotNull] ILoggerFactory logger, 
+            [NotNull] UrlEncoder encoder, 
+            [NotNull] ISystemClock clock) 
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, 
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            var uri = string.Format(ShopifyAuthenticationDefaults.UserInformationEndpoint,
+                properties.Items[ShopifyAuthenticationDefaults.ShopNameAuthenticationProperty]);
+
+            var request = new HttpRequestMessage(HttpMethod.Get, uri);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Add("X-Shopify-Access-Token", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+            
+                throw new HttpRequestException("An error occurred while retrieving the shop profile.");
+            }
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+#if DEBUG
+            // ReSharper disable once UnusedVariable
+            var jsonStr = payload.ToString(Formatting.Indented);
+#endif
+
+            // In Shopify, the customer can modify the scope given to the app. Apps should verify
+            // that the customer is allowing the required scope.
+            var actualScope = tokens.Response["scope"].ToString();
+            var isPersistent = true;
+
+            // If the request was for a "per-user" (i.e. no offline access)
+            if (tokens.Response.TryGetValue("expires_in", out var val))
+            {
+                isPersistent = false;
+
+                var expires = DateTimeOffset.Now.AddSeconds(Convert.ToInt32(val.ToString()));
```

Comment:
Use `CultureInfo.InvariantCulture` for the calls to `ToString()` (I can't tell what the type is because you've used `var` so that might not be valid) and `Convert.ToInt32()` (`int.TryParse()` might be better candidate).